gui/gui.go

```diff
@@ -341,56 +402,10 @@ func (ui *GUI) Run(ctx context.Context) {
 
 				// After three ticks (15 seconds) update cached pool data.
 				if ticks == 3 {
-					var err error
-					work, err := ui.cfg.FetchMinedWork()
-					if err != nil {
-						log.Error(err)
-						continue
-					}
-
-					// Parse and format mined work by the pool.
-					workData := make([]minedWork, 0)
-					for _, work := range work {
-						workData = append(workData, minedWork{
-							BlockHeight: work.Height,
-							BlockURL:    blockURL(ui.cfg.BlockExplorerURL, work.Height),
-							MinedBy:     truncateAccountID(work.MinedBy),
-							Miner:       work.Miner,
-						})
-					}
-
-					// Update mined work cache.
-					ui.minedWorkMtx.Lock()
-					ui.minedWork = workData
-					ui.minedWorkMtx.Unlock()
-
-					quotas, err := ui.cfg.FetchWorkQuotas()
+					err := updateCachedData()
```

Comment:
I think updating cached data should be triggered by incoming new data, would save a lot of db queries.  for example if a new mined work is getting persisted it should trigger a cached update. Perhaps the caches should reside with the pool, they'll be exposed to the gui via the config struct as usual.